static/app/views/automations/detail.tsx

```diff
@@ -86,7 +87,25 @@ export default function AutomationDetail() {
             </Section>
             <Section title={t('Connected Monitors')}>
               <ErrorBoundary mini>
-                <ConnectedMonitorsList monitors={detectors} />
+                <ConnectedMonitorsList
+                  detectors={detectors ?? []}
+                  isLoading={isLoading}
+                  isError={isError}
+                  connectedDetectorIds={new Set(automation.detectorIds)}
```

Comment:
concern here again is the rerenders from the new set every time. might not be worth using a set